src/Exadel.CrazyPrice/Exadel.CrazyPrice.IdentityServer/Configuration/IdentityServerConfiguration.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Exadel.CrazyPrice.Common.Extentions;
+using IdentityServer4.Models;
+using Microsoft.Extensions.Configuration;
+using System.Collections.Generic;
+
+namespace Exadel.CrazyPrice.IdentityServer.Configuration
+{
+    public class IdentityServerConfiguration : IIdentityServerConfiguration
+    {
+        private readonly IConfiguration _config;
+
+        public IdentityServerConfiguration(IConfiguration config)
+        {
+            _config = config;
+        }
+
+        public string[] Origins =>
+            _config.GetArrayString("Auth:Origins");
+
+        public string PolicyName =>
+            _config.GetString("Auth:PolicyName");
+
+        public string IssuerUrl =>
+            _config.GetString("Auth:IssuerUrl");
+
+        public IEnumerable<Client> Clients =>
+            _config.GetSection("Clients").Get<List<Client>>();
+
+        public IEnumerable<ApiScope> ApiScopes =>
+            _config.GetSection("ApiScopes").Get<List<ApiScope>>();
+
+        public IEnumerable<ApiResource> ApiResources =>
+            _config.GetSection("ApiResources").Get<List<ApiResource>>();
+
+        public IEnumerable<IdentityResource> IdentityResources =>
+            _config.GetSection("IdentityResources").Get<List<IdentityResource>>();

```

Comment:
What happens if no such section or invalid format?

Reply:
[Fixed It](https://github.com/yegorl/exadel_discounts_be/pull/24/commits/22d739fdb9e92e5850820466506d6d608135b312)